hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaControllerManagement.java

```diff
@@ -214,10 +210,10 @@ public Target findOrRegisterTargetIfItDoesNotexist(final String controllerId, fi
         final JpaTarget target = targetRepository.findOne(spec);
 
         if (target == null) {
-            final Target result = targetManagement.createTarget(entityFactory.target().create()
+            final Target result = targetRepository.save((JpaTarget) entityFactory.target().create()
```

Comment:
To avoid that one management proxy is calling another one (wrapped retryAble inside a transaction).